File: Tests/Tests/STPPaymentMethodCardWalletVisaCheckoutTest.m
//
//  STPPaymentMethodCardWalletVisaCheckoutTest.m
//  StripeiOS Tests
//
//  Created by Yuki Tokuhiro on 3/9/19.
//  Copyright © 2019 Stripe, Inc. All rights reserved.
//

#import <XCTest/XCTest.h>

#import "STPPaymentMethodCardWalletVisaCheckout.h"
#import "STPFixtures.h"
#import "STPTestUtils.h"

@interface STPPaymentMethodCardWalletVisaCheckoutTest : XCTestCase

@end

@implementation STPPaymentMethodCardWalletVisaCheckoutTest

- (void)testDecodedObjectFromAPIResponseMapping {
    NSDictionary *response = [STPTestUtils jsonNamed:STPTestJSONPaymentMethod][@"card"][@"wallet"][@"visa_checkout"];
    STPPaymentMethodCardWalletVisaCheckout *visaCheckout = [STPPaymentMethodCardWalletVisaCheckout decodedObjectFromAPIResponse:response];
    XCTAssertNotNil(visaCheckout);
    XCTAssertEqualObjects(visaCheckout.name, @"Jenny");
    XCTAssertEqualObjects(visaCheckout.email, @"jenny@example.com");
    XCTAssertNotNil(visaCheckout.billingAddress);
    XCTAssertNotNil(visaCheckout.shippingAddress);
}

@end

File: Stripe/StripeiOSTests/STPPaymentMethodCardWalletVisaCheckoutTest.swift
//  Converted to Swift 5.8.1 by Swiftify v5.8.28463 - https://swiftify.com/
//
//  STPPaymentMethodCardWalletVisaCheckoutTest.m
//  StripeiOS Tests
//
//  Created by Yuki Tokuhiro on 3/9/19.
//  Copyright © 2019 Stripe, Inc. All rights reserved.
//

class STPPaymentMethodCardWalletVisaCheckoutTest: XCTestCase {
    func testDecodedObjectFromAPIResponseMapping() {
        let response = ((STPTestUtils.jsonNamed(STPTestJSONPaymentMethodCard)["card"] as! [AnyHashable: Any])["wallet"] as! [AnyHashable: Any])["visa_checkout"] as? [AnyHashable: Any]
        let visaCheckout = STPPaymentMethodCardWalletVisaCheckout.decodedObject(fromAPIResponse: response)
        XCTAssertNotNil(visaCheckout)
        XCTAssertEqual(visaCheckout?.name, "Jenny")
        XCTAssertEqual(visaCheckout?.email, "jenny@example.com")
        XCTAssertNotNil(visaCheckout?.billingAddress)
        XCTAssertNotNil(visaCheckout?.shippingAddress)
    }
}
